drf_jsonapi/mixins.py

```diff
@@ -213,37 +167,31 @@ class PartialUpdateMixin(ProcessRelationshipsMixin):
     """
 
     def partial_update(self, request, *args, **kwargs):
-        """
-        Build a response for a partial update endpoint
-
-        :param viewset self: This object
-        :param rest_framework.request.Request request: A request object
-        :return: A json response
-        :rtype: Response
-        """
-
-        resource = kwargs.pop('resource', self.get_resource(request, *args, **kwargs))
+        resource = self.get_resource(request, *args, **kwargs)
 
         serializer = self.serializer_class(
             resource,
-            data=request.data['data'],
+            data=request.data["data"],
             only_fields=request.fields,
             partial=True,
             include=request.include,
-            page_size=request.GET.get('page[size]', settings.DEFAULT_PAGE_SIZE),
-            context={'request': request}
+            page_size=request.GET.get(
+                "page[size]",
+                getattr(settings, "DEFAULT_PAGE_SIZE", defaults.DEFAULT_PAGE_SIZE),
+            ),
+            context={"request": request},
         )
 
         if not serializer.is_valid():
-            return self.error_response(
-                Error.parse_validation_errors(serializer.errors)
-            )
+            return self.error_response(Error.parse_validation_errors(serializer.errors))
 
         resource = serializer.save()
 
         # Check for relationships and process them
-        if 'relationships' in request.data['data']:
-            self.process_relationships(request.data['data']['relationships'], resource, request)
+        if "relationships" in request.data["data"]:
+            self.process_relationships(
+                request.data["data"]["relationships"], resource, request
+            )
 
         resource.save()
```

Comment:
@chrisbrantley I noticed that my PATCH calls had 2 UPDATE queries, with the second one seemingly totally unnecessary (essentially just updating modified_at for not real reason). I traced the problem back to this `save()` call. Is there a reason we are saving the object again here? `serializer.save()` seems to also call `save()` on the resource itself.

Reply:
@ddobson FYI Chris won't see this as he doesn't work here any longer.

Reply:
@marcthayer the project is open source and it looks like Chris is using a personal account.

Reply:
@ddobson I think you're right. I believe this was a bit of defensive programming to make sure any changes to the resource object were saved but it is redundant here.  I think it could be safely removed and not cause any problems.

Reply:
@ddobson I would make the change myself but I no longer have write access.